### ISSUE TITLE
Getting EACCES error when starting as normal user. Works as root.

### ISSUE THREAD

**Issue by xtensa** [ethereum/mist#1604]
Have a problem with starting ethereum-wallet as normal user - getting EACCES error. Not getting this issue when starting as root so it is definitely a permission issue. I am not familiar with electron so don't know how to debug it further. If more information needed please give a direction.

#### System information
Gentoo Linux k7 4.2.0-tuxonice #1 SMP x86_64

#### Output log
```python
~# /mnt/4T/eth_blockchain/ethereum-wallet $ ./ethereumwallet --loglevel trace
[2017-01-21 00:34:50.460] [INFO] main - Running in production mode: true
[2017-01-21 00:34:50.492] [TRACE] EthereumNode - Load defaults
[2017-01-21 00:34:50.493] [TRACE] Settings - Load user data /mnt/data/home/rp/.config/Ethereum Wallet/node
[2017-01-21 00:34:50.493] [TRACE] Settings - Load user data /mnt/data/home/rp/.config/Ethereum Wallet/network
[2017-01-21 00:34:50.494] [DEBUG] Sockets - Create socket, id=node-ipc, type=ipc
Secp256k1 bindings are not compiled. Pure JS implementation will be used.
[2017-01-21 00:34:50.642] [INFO] main - Starting in Wallet mode
[2017-01-21 00:34:50.673] [DEBUG] Db - Check that db exists and it's writeable: /mnt/data/home/rp/.config/Ethereum Wallet/mist.lokidb
[2017-01-21 00:34:50.679] [INFO] Db - Loading db: /mnt/data/home/rp/.config/Ethereum Wallet/mist.lokidb
[2017-01-21 00:34:50.688] [INFO] Windows - Creating commonly-used windows
[2017-01-21 00:34:50.689] [TRACE] Windows - Get by type loading
[2017-01-21 00:34:50.689] [INFO] Windows - Create secondary window: loading, owner: notset
[2017-01-21 00:34:50.691] [DEBUG] Windows/loading - Creating browser window
[2017-01-21 00:34:50.720] [DEBUG] Windows/loading - Load URL: file:///mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/interface/index.html#loadingWindow
[2017-01-21 00:34:50.723] [INFO] updateChecker - Check for update...
[2017-01-21 00:34:52.987] [TRACE] ipcProviderBackend - Loaded processors [ 'base',
  'eth_accounts',
  'eth_coinbase',
  'eth_compileSolidity',
  'eth_sendTransaction' ]
[2017-01-21 00:34:53.006] [TRACE] Windows - Get by type main
[2017-01-21 00:34:53.006] [INFO] Windows - Create primary window: main, owner: notset
[2017-01-21 00:34:53.007] [DEBUG] Windows/main - Creating browser window
[2017-01-21 00:34:53.023] [TRACE] Windows - Get by type splash
[2017-01-21 00:34:53.023] [INFO] Windows - Create primary window: splash, owner: notset
[2017-01-21 00:34:53.023] [DEBUG] Windows/splash - Creating browser window
[2017-01-21 00:34:53.033] [DEBUG] Windows/splash - Load URL: file:///mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/interface/index.html#splashScreen_wallet
[2017-01-21 00:34:53.051] [ERROR] main - { Error: spawn EACCES
    at exports._errnoException (util.js:1026:11)
    at ChildProcess.spawn (internal/child_process.js:313:11)
    at exports.spawn (child_process.js:392:9)
    at Object.exports.execFile (child_process.js:155:15)
    at Object.module.(anonymous function) [as execFile] (ELECTRON_ASAR.js:195:20)
    at checkSystemd (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/os-timesync/index.js:41:18)
    at checkLinux (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/os-timesync/index.js:55:5)
    at Object.checkEnabled (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/os-timesync/index.js:81:16)
    at onReady (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/main.js:254:14)
    at tryCatcher (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/bluebird/js/release/promise.js:510:31)
    at Promise._settlePromise (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/bluebird/js/release/promise.js:567:18)
    at Promise._settlePromise0 (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/bluebird/js/release/promise.js:612:10)
    at Promise._settlePromises (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/bluebird/js/release/promise.js:691:18)
    at Async._drainQueue (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/bluebird/js/release/async.js:138:16)
    at Async._drainQueues (/mnt/4T/eth_blockchain/ethereum-wallet/resources/app.asar/node_modules/bluebird/js/release/async.js:148:10) code: 'EACCES', errno: 'EACCES', syscall: 'spawn' }
[2017-01-21 00:34:53.076] [INFO] main - Defer quitting until sockets and node are shut down
[2017-01-21 00:34:53.076] [INFO] Sockets - Destroy all sockets
[2017-01-21 00:34:53.076] [DEBUG] Sockets - Remove socket, id=node-ipc
[2017-01-21 00:34:53.077] [INFO] Sockets/node-ipc - Disconnecting...
[2017-01-21 00:34:53.079] [ERROR] main - Error shutting down sockets
[2017-01-21 00:34:53.220] [DEBUG] Windows - Set window id 3
[2017-01-21 00:34:53.221] [TRACE] Windows - Set window id=3, type=splash
[2017-01-21 00:34:53.376] [INFO] ipcCommunicator - Backend language set to:  en-US
[2017-01-21 00:34:53.397] [DEBUG] ipcProviderBackend - Create new socket connection, id=3
[2017-01-21 00:34:53.397] [DEBUG] Sockets - Create socket, id=3, type=ipc
[2017-01-21 00:34:53.398] [DEBUG] ipcProviderBackend - Connecting socket 3
[2017-01-21 00:34:53.580] [INFO] main - About to quit...
[2017-01-21 00:34:53.581] [DEBUG] Windows/main - Closed
[2017-01-21 00:34:53.582] [DEBUG] Windows - Removing window from list: main
[2017-01-21 00:34:53.582] [INFO] Windows - All primary windows closed/invisible, so quitting app...
[2017-01-21 00:34:53.583] [DEBUG] Windows/splash - Closed
[2017-01-21 00:34:53.583] [DEBUG] Windows - Removing window from list: splash
[2017-01-21 00:34:53.583] [INFO] Windows - All primary windows closed/invisible, so quitting app...
[2017-01-21 00:34:53.984] [INFO] updateChecker - App is up-to-date.
[2017-01-21 00:34:54.124] [DEBUG] Windows/loading - Content loaded, id: undefined
[2017-01-21 00:34:54.132] [DEBUG] Windows/loading - Closed
[2017-01-21 00:34:54.133] [DEBUG] Windows - Removing window from list: loading
[2017-01-21 00:34:54.133] [INFO] Windows - All primary windows closed/invisible, so quitting app...
```

**Comment by xtensa** [comment]
A week passed and no one even assigned...

Anyway I managed to resolve problem by unpacking app.asar and investigating what the program is trying to do. As I see it is relying on timedatectl call to check if system is configured with NTP. "timedatectl" should not be installed on every linux system. In my case I am not using systemd and I don't have timedatectl. However, I have NTP configured and it working correctly. On the other hand, ethereum-wallet started as a root but warning message was displayed that system is not using NTP time. The workaround is simple - i created executable script called timedatectl and added it to the PATH. The content of this script is trivial:

`#!/bin/bash

# this script is meant to mimic timedatectl program provided by systemd package

echo "Network time on: yes"
`

Now ethereum/mist starting properly.

**Comment by frozeman** [comment]
Thanks for the issue. Sorry for the delay we are working on other things right now. So issues are on th backlog and some of us are traveling right now. 

This issue is in the ntp check package we use I guess or in go Ethereum itself. 
@fjl can you have a look?

**Comment by fjl** [comment]
Please upgrade to os-timesync v1.0.7

**Comment by luclu** [comment]
PR including `os-timesync` `1.0.7`: https://github.com/ethereum/mist/pull/1622

**Comment by luclu** [comment]
@xtensa could you please check whether this issue is resolved in `v0.8.9`?
